pkg/ingester/ingester.go

```diff
@@ -2011,7 +2126,7 @@ func (i *Ingester) compactBlocks(ctx context.Context, force bool, allowed *util.
 
 		default:
 			reason = "regular"
-			err = userDB.Compact()
+			err = userDB.Compact(time.Now().Add(-i.cfg.EphemeralSeriesRetentionPeriod))
```

Comment:
Have you considered storing `EphemeralSeriesRetentionPeriod` in `userTSDB` instead? It's a bit weird reading the function call here. It makes you think it just compacts the ephemeral TSDB, while it compacts both.

Reply:
Done.